File: string/check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.ts
function digitCount(num: string): boolean {
    const count = Array<number>(10).fill(0)

    for (let c of num) {
        count[parseInt(c)]++;
    }

    for (let i = 0; i < num.length; i++) {
        if (count[i] !== parseInt(num[i])) {
            return false;
        }
    }

    return true;
};
// autobiographical numbers, introduced by Da Vinci and their is only certain set of numbers which satisfy the condition.

File: string/check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.go
package checkifnumberhasequaldigitcountanddigitvalue

func digitCount(num string) bool {
	var count [10]int

	for _, c := range num {
		count[int(c-'0')]++
	}

	for i := 0; i < len(num); i++ {
		if count[i] != int(num[i]-'0') {
			return false
		}
	}
	return true
}
